File: cpp_solutions/78.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<vector<int>> res;
        int n = nums.size();
        for (int i = 0; i < (1 << n); ++i) {
            vector<int> temp;
            for (int j = 0; j < n; ++j) {
                if (i & (1 << j)) temp.push_back(nums[j]);
            }
            res.push_back(temp);
        }
        return res;
    }
};

File: rust_solutions/78.rs
impl Solution {
    pub fn subsets(nums: Vec<i32>) -> Vec<Vec<i32>> {
        let n = nums.len();
        let mut res: Vec<Vec<i32>> = vec![];

        for i in 0..2_u32.pow(n as u32) {
            let mut subset = vec![];
            for j in 0..n {
                if (i & (1 << j)) > 0 {
                    subset.push(nums[j]);
                }
            }
            res.push(subset);
        }
        return res;
    }
}
